core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -176,11 +177,8 @@ protected function prepare($objPage)
 
 		$this->Template->sections = $arrCustomSections;
 
-		// Mark RTL languages (see #7171)
-		if ($GLOBALS['TL_LANG']['MSC']['textDirection'] == 'rtl')
-		{
-			$this->Template->isRTL = true;
-		}
+		// Mark RTL languages (see #7171, #3360)
+		$this->Template->isRTL = 'rtl' === ($GLOBALS['TL_LANG']['MSC']['textDirection'] ?? Cosmo::create($GLOBALS['TL_LANGUAGE'])->direction());
```

Comment:
_Note:_ Intl is used as a fallback, in case anyone is forcing the `MSC.textDirection` via `contao/languages/…/default.php`.

Reply:
```suggestion
		$this->Template->isRTL = 'rtl' === ($GLOBALS['TL_LANG']['MSC']['textDirection'] ?? ((\ResourceBundle::create($GLOBALS['TL_LANGUAGE'], 'ICUDATA', true)['layout']['characters'] ?? null) === 'right-to-left' ? 'rtl' : 'ltr'));
```